examples/metalk8s-solution-example/ui/src/services/k8s/api.js

```diff
@@ -84,26 +143,34 @@ export async function updateCustomResource(body, namespaces, name) {
   }
 }
 
-export async function getSolutionNamespaces() {
+export async function updateVersionServer(body, namespaces, name) {
   try {
-    return await coreV1.listNamespace(
-      null,
-      null,
-      null,
-      null,
-      `${LABEL_PART_OF}=example-solution`
+    return await customObjects.patchNamespacedCustomObject(
+      'example-solution.metalk8s.scality.com',
+      'v1alpha1',
+      namespaces,
+      'versionservers',
+      name,
+      body,
+      {
+        headers: {
+          'Content-Type': 'application/merge-patch+json'
+        }
+      }
     );
   } catch (error) {
     return { error };
   }
 }
 
-export async function getSolutionDeployment(version) {
+export async function getOperatorDeployments(namespaces, name) {
   try {
-    return await appsV1Api.listDeploymentForAllNamespaces(
+    return await appsV1Api.listNamespacedDeployment(
```

Comment:
readNamespacedDeployment works as well. However, when the deployment does not exist, it pops a 404 error whereas listNamespacedDeployment return a 200 with an empty list. We use it to populate the environment state, so we do not want to deal with the error.

Reply:
Mmh, you're not using this Deployment to fill an array in the state, this is, as you said, an optional attribute for each Environment. So it could(should) try to get the object and return nothing if a caught error has `status_code == 404`, failing otherwise.